noir-projects/noir-protocol-circuits/crates/rollup-lib/src/base/base_rollup_inputs.nr

```diff
@@ -128,10 +128,8 @@ impl BaseRollupInputs {
             self.check_nullifier_tree_non_membership_and_insert_to_tree();
 
         // Inject protocol update requests for deducting tx_fee from fee_payer's balance
-        let gas_fees = self.constants.global_variables.gas_fees;
-        let transaction_fee = self.kernel_data.public_inputs.compute_transaction_fee(gas_fees);
         let all_public_data_update_requests =
-            self.calculate_all_public_data_update_requests(transaction_fee);
+            self.calculate_all_public_data_update_requests(self.transaction_fee);
```

Comment:
`transaction_fee` is either computed in `private_base_rollup` or in the AVM and passed in from `public_base_rollup`.